### PR TITLE
ZK-3792: enter am but Datebox converts it to PM

### PR DESCRIPTION
**Pull request by klyvechen**

### CODE REVIEW

File: zul/src/archive/web/js/zul/db/datefmt.js [hunk 1]
```diff
@@ -371,7 +371,7 @@ zk.fmt.Date = {
 					if (!hasHour1)
 						break;
 					if (!token) return; //failed
-					isAM = token.startsWith(localizedSymbols.APM[0]);
+					isAM = token.toUpperCase().startsWith(localizedSymbols.APM[0]);
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/db/datefmt.js]: Both token and APM[0]`.toUpperCase()` are safer

**Comment by scribetw** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/zkoss/zk/1127)*

File: zul/src/archive/web/js/zul/db/datefmt.js (hunk 1 above)
Comment (klyvechen) [reply zul/src/archive/web/js/zul/db/datefmt.js]: ok

File: zktest/src/archive/test2/B85-ZK-3792.zul [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+                
+        History:
+                Thu May 31 10:56:07 CST 2018, Created by klyve
+
+Copyright (C) 2018 Potix Corporation. All Rights Reserved.
+
+-->
+
+<zk>
+	<datebox format="KK:mm a" lenient="true"/>
```
Comment (scribetw) [inline zktest/src/archive/test2/B85-ZK-3792.zul]: No instructions, testers don't know what to test and what to expect

Comment (klyvechen) [reply zktest/src/archive/test2/B85-ZK-3792.zul]: ok

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**